File: test/codeBase-js/fsm.color/tmp/Red.js
// version: V1

const util = require("util")
const Red = function (context) {
	this.context = context
}

// Partial evaluate
require("./Red.evaluate")(Red)

// ----------
// State Logic Implement Below
// ----------

Red.prototype.OnUpdate = function(stateData, dt) {	
	// Logic Code Here

	return this._evaluate(stateData)
};

Red.prototype.OnEnter = function(stateData) {
	// Logic Code Here
	console.log("Red - Enter")
};

Red.prototype.OnExit = function(stateData) {
	// Logic Code Here
	console.log("Red - Exit")
};

Red.prototype.OnEvent = function(stateData, args) {
	// Logic Code Here
	console.log("Red - On Event ", args)
};

module.exports = Red

File: codeBase/cs/FsmCodeBase/FsmCodeBase/Test/Logic/Red.cs
using System;
using FsmCodeBase;

namespace FsmCodeBase.Test
{
    partial class Red : BaseState
    {
        public override string GetName()
        {
            return "Red";
        }

        public override void OnEnter(StateMachine context)
        {
            Console.WriteLine("[OnEnter] Red");
            base.OnEnter(context);
        }

        public override string OnUpdate(float dt, StateMachine context)
        {
            return base.OnUpdate(dt, context);
        }

        public override void OnEvent(string name, object args, StateMachine context)
        {
            base.OnEvent(name, args, context);
        }

        public override void OnExit(StateMachine context)
        {
            Console.WriteLine("[OnExit] Red");
            base.OnExit(context);
        }
    }
}
